TreeView: Maximum call stack size exceeded error

Max call stack error when there's too many nodes in a TreeView

Steps to Reproduce:

1. Install JSON TextMate [extension](https://marketplace.visualstudio.com/items?itemName=RedCMD.tmlanguage-syntax-highlighter)
2. Load a file that causes a TextMate grammar to match many many rules
  `typename::std::add_pointer<typename::std::add_pointer<typename::std::add_pointer<typename::std::add_pointer<typename::std::add_pointer<typename::std::add_pointer<typename::std::add_pointer<typename::std::add_pointer<typename::std::add_pointer<int>::type>::type>::type>::type>::type>::type>::type>::type>::type ptr{nullptr};`
works well in a CPP file
3. right click => `Show TextMate Calling Stack`
4. wait for file to be parsed
5. wait even longer for VSCode to attempt to render the tree (notice memory usage going up)
6. Max Call Stack error

```rust
  ERR Maximum call stack size exceeded: RangeError: Maximum call stack size exceeded
    at E.U (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:191:12436)
    at E.splice (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:191:11388)
    at vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:4:9067
    at Array.forEach (<anonymous>)
    at t.splice (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:4:9054)
    at vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:234:8098
    at A.bufferEvents (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:100:3899)
    at te.splice (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:234:8074)
    at te.splice (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:628:64439)
    at f.t (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:187:27391)
    at f.splice (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:187:25598)
    at L.l (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:187:34140)
    at L.setChildren (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:187:33684)
    at C.setChildren (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:629:6619)
    at Re.J (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:629:23716)
    at Re.C (vscode-file://vscode-app/e:/VSCode-win32-x64/resources/app/out/vs/workbench/workbench.desktop.main.js:629:20783)
```
Seems TreeView cant expand more than 170k nesting nodes all at once

related: #61524

- VS Code Version: 1.93.1
- OS Version: Windows 11

I get an exception in the extension when I try to use it. 

```
workbench.web.main.internal.js:3109  Error: No grammar provided for <undefined>
    at collectReferencesOfReference (/home/node/.vscode-remote/extensions/redcmd.tmlanguage-syntax-highlighter-2.5.0/out/textmate/grammar/grammarDependencies.js:102:19)
    at ScopeDependencyProcessor.processQueue (/home/node/.vscode-remote/extensions/redcmd.tmlanguage-syntax-highlighter-2.5.0/out/textmate/grammar/grammarDependencies.js:71:13)
    at Registry._loadGrammar (/home/node/.vscode-remote/extensions/redcmd.tmlanguage-syntax-highlighter-2.5.0/out/textmate/main.js:81:33)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at tokenizeFile (/home/node/.vscode-remote/extensions/redcmd.tmlanguage-syntax-highlighter-2.5.0/out/TextMate.js:204:21)
    at Object.getChildren (/home/node/.vscode-remote/extensions/redcmd.tmlanguage-syntax-highlighter-2.5.0/out/Providers/TreeDataProvider.js:34:23)
    at vy.Y (file:///vscode/bin/linux-x64/d78a74bcdfad14d5d3b1b782f87255d802b57511-insider/out/vs/workbench/api/node/extensionHostProcess.js:156:13744)
    at vy.getChildren (file:///vscode/bin/linux-x64/d78a74bcdfad14d5d3b1b782f87255d802b57511-insider/out/vs/workbench/api/node/extensionHostProcess.js:156:10258)
```

--- a/src/vs/base/browser/ui/tree/asyncDataTree.ts
+++ b/src/vs/base/browser/ui/tree/asyncDataTree.ts
@@ -25,6 +25,7 @@ import { CancellationToken, CancellationTokenSource } from '../../../common/canc
 import { IObjectTreeModel } from './objectTreeModel.js';
 import { IContextViewProvider } from '../contextview/contextview.js';
 import { FuzzyScore } from '../../../common/filters.js';
+import { splice } from '../../../common/arrays.js';
 
 interface IAsyncDataTreeNode<TInput, T> {
 	element: TInput | T;
@@ -1445,7 +1446,7 @@ export class AsyncDataTree<TInput, T, TFilterData = void> implements IDisposable
 			this.nodes.set(child.element as T, child);
 		}
 
-		node.children.splice(0, node.children.length, ...children);
+		splice(node.children, 0, node.children.length, children);
 
 		// TODO@joao this doesn't take filter into account
 		if (node !== this.root && this.autoExpandSingleChildren && children.length === 1 && childrenToRefresh.length === 0) {

